src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -1435,13 +1435,23 @@ private static LambdaConversionResult IsAnonymousFunctionCompatibleWithDelegate(
 
                 if (anonymousFunction.HasExplicitlyTypedParameterList)
                 {
+                    var lambdaSymbol = anonymousFunction.TemporaryLambdaSymbol;
+
+                    Debug.Assert(lambdaSymbol is not null);
+
                     for (int p = 0; p < delegateParameters.Length; ++p)
                     {
                         if (delegateParameters[p].RefKind != anonymousFunction.RefKind(p) ||
                             !delegateParameters[p].Type.Equals(anonymousFunction.ParameterType(p), TypeCompareKind.AllIgnoreOptions))
                         {
                             return LambdaConversionResult.MismatchedParameterType;
                         }
+
+                        if (lambdaSymbol.Parameters[p].HasExplicitDefaultValue &&
+                            delegateParameters[p].ExplicitDefaultConstantValue != lambdaSymbol.Parameters[p].ExplicitDefaultConstantValue)
```

Comment:
>lambdaSymbol.Parameters[p].ExplicitDefaultConstantValue [](http://example.com/codeflow?start=82&length=55)

Can the default values be stored on the `FunctionType` rather than introducing a `TemporaryLambdaSymbol`?

Reply:
`FunctionType` is not calculated in all cases.

Reply:
Unfortunately it seems like not, though it would be nice to have an alternate solution to the `TemporaryLambdaSymbol` if possible